Sources/Build/BuildOperationBuildSystemDelegateHandler.swift

```diff
@@ -465,6 +470,61 @@ public final class BuildExecutionContext {
     }
 }
 
+final class WriteAuxiliaryFileCommand: CustomLLBuildCommand {
+    override func getSignature(_ command: SPMLLBuild.Command) -> [UInt8] {
+        guard let buildDescription = self.context.buildDescription else {
+            return []
+        }
+        guard let tool = buildDescription.copyCommands[command.name] else {
+            return []
+        }
+
+        let encoder = JSONEncoder.makeWithDefaults()
+        var hash = Data()
+        hash += try! encoder.encode(tool.inputs)
+        hash += try! encoder.encode(tool.outputs)
+        return [UInt8](hash)
+    }
+
+    override func execute(
+        _ command: SPMLLBuild.Command,
+        _: SPMLLBuild.BuildSystemCommandInterface
+    ) -> Bool {
+        do {
+            guard let buildDescription = self.context.buildDescription else {
+                throw InternalError("unknown build description")
+            }
+            guard let tool = buildDescription.writeCommands[command.name] else {
+                throw StringError("command \(command.name) not registered")
+            }
+
+            guard let output = tool.outputs.first, output.kind == .file else {
+                throw StringError("invalid output path")
+            }
+            let outputFilePath = try AbsolutePath(validating: output.name)
+
+            try self.context.fileSystem.writeFileContents(outputFilePath, string: getFileContents(tool: tool))
+            return true
+        } catch {
+            self.context.observabilityScope.emit(error: "failed to write auxiliary file: \(error.interpolationDescription)")
+            return false
+        }
+    }
+
+    func getFileContents(tool: WriteAuxiliaryFile) throws -> String {
+        guard tool.inputs.first?.kind == .virtual, let generatedFileType = tool.inputs.first?.name.dropFirst().dropLast() else {
+            throw StringError("invalid inputs")
+        }
+
+        switch generatedFileType {
+        case WriteAuxiliary.LinkFileList.name:
+            return try WriteAuxiliary.LinkFileList.getFileContents(inputs: Array(tool.inputs.dropFirst()))
+        default:
+            throw InternalError("unhandled generated file type '\(generatedFileType)'")
```

Comment:
Could this error include a bit more information, for example a path of this generated file?

Reply:
Don't think we should do it on a per error basis, but we could add it [here](https://github.com/apple/swift-package-manager/pull/6606/files#diff-d817929f28bcb3a115bdf6b9226bbb2736513292b73baefcfd4f9b164fc52281R509) maybe

Reply:
Actually, that isn't possible since for most of it we don't know the output file name, yet.

Reply:
Broke up the callee now so that we can inject the file name once we have it.